nemo_text_processing/text_normalization/normalize_with_audio.py

```diff
@@ -496,34 +502,50 @@ def __process_batch(batch_idx: int, batch: List[str], dir_name: str):
         )
 
         if os.path.exists(args.text):
-            with open(args.text, 'r') as f:
-                args.text = f.read().strip()
-        normalized_texts = normalizer.normalize(
-            text=args.text,
-            verbose=args.verbose,
-            n_tagged=args.n_tagged,
-            punct_post_process=not args.no_punct_post_process,
-        )
+            metadata = dict()
+            all_audio_data = []
+            all_input_text = []
+            all_targets = []
+            data = Path(args.text).parent
+            with open(args.text) as f:
+                for line in f:
+                    filename, *_, text = line.strip().split('|')
+                    metadata[filename] = text
+                    full_path = data / "wavs" / f'{filename}.wav'
+                    all_audio_data.append(str(full_path))
+                    all_input_text.append(text)
+                    all_targets.append([])
+
+        for idx, text in enumerate(all_input_text):
+            normalized_texts = normalizer.normalize(
+                text=text,
+                verbose=args.verbose,
+                n_tagged=args.n_tagged,
+                punct_post_process=not args.no_punct_post_process,
+            )
+
+            if not normalizer.lm:
+                normalized_texts = set(normalized_texts)
 
-        if not normalizer.lm:
-            normalized_texts = set(normalized_texts)
-        if args.audio_data:
-            asr_model = get_asr_model(args.model)
-            pred_text = asr_model.transcribe([args.audio_data])[0]
+            pred_text = asr_model.transcribe([all_audio_data[idx]])[0]
             normalized_text, cer = normalizer.select_best_match(
                 normalized_texts=normalized_texts,
                 pred_text=pred_text,
-                input_text=args.text,
+                input_text=text,
                 verbose=args.verbose,
                 remove_punct=not args.no_remove_punct_for_cer,
                 cer_threshold=args.cer_threshold,
             )
-            print(f"Transcript: {pred_text}")
-            print(f"Normalized: {normalized_text}")
-        else:
-            print("Normalization options:")
-            for norm_text in normalized_texts:
-                print(norm_text)
+            # print(f"Transcript: {pred_text}")
+            # print(f"Normalized: {normalized_text}")
+            all_norm_output.append(normalized_text)
+
+            if (idx + 1) % 50 == 0:
+                with open(args.text + 'norm', "a") as f_out:
+                    for out_idx, line in enumerate(all_norm_output):
+                        f_out.write(all_meta[idx-10 + out_idx+1] + '|' + line + "\n")
+                all_norm_output = []
+
```

Comment:
is this checkpointing? it doesn't handle metadata shorter than 50 lines and will skip the last bucket of lines in a large metadata.